tests/unit/dom/type-in-test.js

```diff
@@ -0,0 +1,133 @@
+import { module, test } from 'qunit';
+import { typeIn, setupContext, teardownContext } from '@ember/test-helpers';
+import { buildInstrumentedElement } from '../../helpers/events';
+import { isIE11 } from '../../helpers/browser-detect';
+import hasEmberVersion from 'ember-test-helpers/has-ember-version';
+
+let expectedEvents = [
+  'focus',
```

Comment:
After some investigation, I think the order of the events may be wrong. Check this jsbin: https://jsbin.com/zitazuxabe/edit?html,js,console,output

Seems that the sequence when typing "abc" is:
```
keydown | e.target.value is ""
keypress | e.target.value is ""
input | e.target.value is "a"
keyup | e.target.value is "a"
keydown | e.target.value is " a"
keypress | e.target.value is "a"
input | e.target.value is "ab"
keyup | e.target.value is "ab"
keydown | e.target.value is " ab"
keypress | e.target.value is "ab"
input | e.target.value is "abc"
keyup | e.target.value is "abc"
change | e.target.value is "abc"
```

Changes I request:

- `input` fires before `keyup`
- It's important that `keydown` and `keypress` are fired **before** the input gets the new value, and the rest are fired after
- The `change` event should only be fired once at the end.

Reply:
Done

Reply:
Can you add this JSBin as an inline comment? It sure will make "reality" testing easier over time...